sdk/cmake/host-tools.cmake

```diff
@@ -26,6 +26,48 @@ function(setup_host_tools)
         set(HOST_MODULE_SUFFIX ".so")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH x86) # host-tools are not happy compiling for x64...
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL i686 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL i386)
```

Comment:
You won't see `i386` on a real linux distro these days, it's `i686` most of the time or `i586` on some old Debians
Windows has it set to "x86" https://superuser.com/questions/305901/possible-values-of-processor-architecture/305902

Reply:
use MATCH i[3456]86 ?

Reply:
"REGEX MATCH" is only applicable to `string` command in CMake, so this will need one more variable :)

Reply:
Oh yes, this is MATCHES. Then the wole thing would be
`if (CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "[iI][3456]86")` I guess.